lib/nanoc/base/entities/configuration.rb

```diff
@@ -33,11 +33,19 @@ class Configuration
       string_pattern_type: 'glob',
     }.freeze
 
-    contract Hash => C::Any
+    # @return [String, nil] The active environment for the configuration
+    attr_reader :env
+
+    # Configuration environments property key
+    ENVIRONMENTS = :environments
+
+    contract Hash, C::Maybe[String] => C::Any
     # Creates a new configuration with the given hash.
     #
     # @param [Hash] hash The actual configuration hash
-    def initialize(hash = {})
+    # @param [String, nil] env The active environment for this configuration
+    def initialize(hash = {}, env = nil)
```

Comment:
The constructor of this class is becoming not so obvious at this point. Can you use named arguments rather than default values? (e.g. `def initialize(hash: {}, env: nil)`)